src/pages/academy/components/utility.ts

```diff
@@ -36,17 +37,32 @@ export const handleTag = (tag_name) => {
 }
 
 export const useDataFilter = (data) => {
-    const { is_eu, is_uk } = useCountryRule()
-    let filtered_data = data
+    const [filtered_data, setFilteredData] = React.useState(null)
+    const { is_eu, is_uk, is_loading } = useCountryRule()
+
+    React.useEffect(() => {
+        if (!is_loading) {
+            if (is_eu) {
+                setFilteredData(
+                    data.filter(
+                        (item) =>
+                            item.visibility !== 'hide_for_eu' &&
+                            item.visibility !== 'hide_for_eu_uk',
+                    ),
+                )
+            } else if (is_uk) {
```

Comment:
Looks like duplicated logic ?

Reply:
One of them contains `item.visibility !== 'hide_for_eu' &&` an another one `item.visibility !== 'hide_for_uk' &&`.

Reply:
Create a function and pass the values via argument, `hide_for_eu` `hide_for_eu_uk` `hide_for_uk`. @mohammad-rework